packages/next/src/lib/load-custom-routes.ts

```diff
@@ -525,7 +525,9 @@ function processRoutes<T>(
           newRoutes.push({
             ...r,
             destination,
-            source: `${srcBasePath}/${item.locale}${r.source}`,
+            source: `${srcBasePath}/${item.locale}${
+              r.source === '/' && !config.trailingSlash ? '' : r.source
```

Comment:
Before this special change, if `source` was `/` and `locale` was `en`, it was pushing

```js
{ source: '/en/', ... }
```

to `newRoutes`, but because of the trailing slash in `/en/`, it was causing the route not to match when you access `/`.

On the other hand, couple lines below, the logic for `nextInternalLocale` already handled the special case for `/`

https://github.com/vercel/next.js/blob/2a8793866740b9234ee6d94f939b931a61d78eb8/packages/next/src/lib/load-custom-routes.ts#L535-L539

So I simply ported this. This will now push `/en` (without trailing slash) to `newRoutes` as desired.

```js
{ source: '/en', ... }
```